x/ethbridge/client/cli/tx.go

```diff
@@ -1,56 +1,57 @@
 package cli
 
 import (
-	"bufio"
 	"regexp"
 	"strconv"
 
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
-
-	"github.com/cosmos/cosmos-sdk/client/context"
-	"github.com/cosmos/cosmos-sdk/codec"
-	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 
 	"github.com/Sifchain/sifnode/x/ethbridge/types"
 )
 
 // GetCmdCreateEthBridgeClaim is the CLI command for creating a claim on an ethereum prophecy
 //nolint:lll
-func GetCmdCreateEthBridgeClaim(cdc *codec.Codec) *cobra.Command {
+func GetCmdCreateEthBridgeClaim() *cobra.Command {
 	return &cobra.Command{
 		Use:   "create-claim [bridge-registry-contract] [nonce] [symbol] [ethereum-sender-address] [cosmos-receiver-address] [validator-address] [amount] [claim-type] --ethereum-chain-id [ethereum-chain-id] --token-contract-address [token-contract-address]",
 		Short: "create a claim on an ethereum prophecy",
 		Args:  cobra.ExactArgs(8),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
 
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
-			inBuf := bufio.NewReader(cmd.InOrStdin())
-			txBldr := authtypes.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			flags := cmd.Flags()
+
+			ethereumChainID, err := flags.GetInt64(types.FlagEthereumChainID)
+			if err != nil {
+				return err
+			}
 
-			ethereumChainIDString := viper.GetString(types.FlagEthereumChainID)
-			ethereumChainID, err := strconv.Atoi(ethereumChainIDString)
+			tokenContractString, err := flags.GetString(types.FlagTokenContractAddr)
 			if err != nil {
 				return err
 			}
 
-			tokenContractString := viper.GetString(types.FlagTokenContractAddr)
 			if !common.IsHexAddress(tokenContractString) {
 				return errors.Errorf("invalid [token-contract-address]: %s", tokenContractString)
 			}
+
 			tokenContract := types.NewEthereumAddress(tokenContractString)
 
 			if !common.IsHexAddress(args[0]) {
 				return errors.Errorf("invalid [bridge-registry-contract]: %s", args[0])
 			}
 			bridgeContract := types.NewEthereumAddress(args[0])
 
-			nonce, err := strconv.Atoi(args[1])
+			nonce, err := strconv.ParseInt(args[1], 10, 64)
```

Comment:
Why are we using ParseInt instead of ATOI?

Reply:
Reason for using Parseint instead of Atoi is parseint stores result in larger data container.